test/topology_custom/test_hints.py

```diff
@@ -77,3 +78,62 @@ async def test_limited_concurrency_of_writes(manager: ManagerClient):
         except NoHostAvailable as e:
             for _, err in e.errors.items():
                 assert err.summary == "Coordinator node overloaded" and re.match(r"Too many in flight hints: \d+", err.message)
+
+@pytest.mark.asyncio
+async def test_sync_point(manager: ManagerClient):
+    """
+    We want to verify that the sync point API is compliant with its design.
+    This test concerns one particular aspect of it: Scylla should create a sync point
+    for ALL nodes if the parameter `target_hosts` of a request is empty, not just
+    live nodes.
+    """
+    node_count = 3
+    [node1, node2, node3] = await manager.servers_add(node_count)
+
+    cql = manager.get_cql()
+    await cql.run_async("CREATE KEYSPACE ks WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 3}")
+    await cql.run_async("CREATE TABLE ks.t (pk int primary key, v int)")
+
+    # Creates a sync point for ALL hosts.
+    def create_sync_point(node: ServerInfo) -> str:
+        return requests.post(f"http://{node.ip_addr}:10000/hinted_handoff/sync_point/").json()
+
+    def await_sync_point(node: ServerInfo, sync_point: str, timeout: int) -> bool:
+        params = {
+            "id": sync_point,
+            "timeout": str(timeout)
+        }
+
+        response = requests.get(f"http://{node.ip_addr}:10000/hinted_handoff/sync_point", params=params).json()
+        match response:
+            case "IN_PROGRESS":
+                return False
+            case "DONE":
+                return True
+            case _:
+                pytest.fail(f"Unexpected response from the server: {response}")
+
+    await manager.server_stop_gracefully(node2.server_id)
+    await manager.server_stop_gracefully(node3.server_id)
+
+    await manager.server_not_sees_other_server(node1.ip_addr, node2.ip_addr)
+    await manager.server_not_sees_other_server(node1.ip_addr, node3.ip_addr)
+
+    mutation_count = 5
+    for primary_key in range(mutation_count):
+        await cql.run_async(SimpleStatement(f"INSERT INTO ks.t (pk, v) VALUES ({primary_key}, {primary_key})", consistency_level=ConsistencyLevel.ONE))
+
+    # Give the test 5 seconds to apply the mutations. They need to be applied to hinted handoff's commitlog
+    # before we create the sync point. Otherwise, the sync point will correspond to no hints at all.
+    await asyncio.sleep(5)
```

Comment:
Here, you could probably wait until the `storage_proxy_size_of_hints_in_progress` (or something similar) metric drops down to 0 instead of waiting.